src/Symfony/Component/Ldap/Adapter/ExtLdap/ResultIterator.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap\Adapter\ExtLdap;
+
+use Symfony\Component\Ldap\Entry;
+
+/**
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class ResultIterator implements \Iterator
+{
+    private $connection;
+    private $search;
+    private $current;
+    private $key;
+
+    /**
+     * Constructor.
+     *
+     * @param Connection $connection
+     * @param Query      $search
+     */
```

Comment:
The PHPDoc here overrides the class used for the Connection instance. This helps for the typehint, as we actually do not use the interface here (this class is specific to the implementation).